src/bitmessagekivy/baseclass/maildetail.py

```diff
@@ -0,0 +1,256 @@
+)
+from bitmessagekivy.baseclass.popup import SenderDetailPopup
+from bitmessagekivy.get_platform import platform
+from helper_sql import sqlQuery
+from helper_sent import delete, retrieve_message_details
+from helper_inbox import trash
+
+
+class OneLineListTitle(OneLineListItem):
+    """OneLineListTitle class for kivy Ui"""
+    __events__ = ('on_long_press', )
+    long_press_time = NumericProperty(1)
+
+    def on_state(self, instance, value):
+        """On state"""
+        if value == 'down':
+            lpt = self.long_press_time
+            self._clockev = Clock.schedule_once(self._do_long_press, lpt)
+        else:
+            self._clockev.cancel()
+
+    def _do_long_press(self, dt):
+        """Do long press"""
+        self.dispatch('on_long_press')
+
+    def on_long_press(self, *largs):
+        """On long press"""
+        self.copymessageTitle(self.text)
+
+    def copymessageTitle(self, title_text):
+        """this method is for displaying dialog box"""
+        self.title_text = title_text
+        width = .8 if platform == 'android' else .55
+        self.dialog_box = MDDialog(
+            text=title_text,
+            size_hint=(width, .25),
+            buttons=[
+                MDFlatButton(
+                    text="Copy", on_release=self.callback_for_copy_title
+                ),
+                MDFlatButton(
+                    text="Cancel", on_release=self.callback_for_copy_title,
+                ),
+            ],)
+        self.dialog_box.open()
+
+    def callback_for_copy_title(self, instance):
+        """Callback of alert box"""
+        if instance.text == 'Copy':
+            Clipboard.copy(self.title_text)
+        self.dialog_box.dismiss()
+        toast(instance.text)
+
+
+class IconRightSampleWidget(IRightBodyTouch, MDIconButton):
+    """IconRightSampleWidget class for kivy Ui"""
+
+
+class MailDetail(Screen):  # pylint: disable=too-many-instance-attributes
+    """MailDetail Screen class for kivy Ui"""
+
+    to_addr = StringProperty()
+    from_addr = StringProperty()
+    subject = StringProperty()
+    message = StringProperty()
+    status = StringProperty()
+    page_type = StringProperty()
+    time_tag = StringProperty()
+    avatarImg = StringProperty()
+    no_subject = '(no subject)'
+
+    def __init__(self, *args, **kwargs):
+        """Mail Details method"""
+        super(MailDetail, self).__init__(*args, **kwargs)
+        self.kivy_running_app = App.get_running_app()
+        self.kivy_state = self.kivy_running_app.kivy_state_obj
+        Clock.schedule_once(self.init_ui, 0)
+
+    def init_ui(self, dt=0):
+        """Clock Schdule for method MailDetail mails"""
+        self.page_type = self.kivy_state.detailPageType if self.kivy_state.detailPageType else ''
+        try:
+            if self.kivy_state.detailPageType in ('sent', 'draft'):
+                data = retrieve_message_details(self.kivy_state.mail_id)
+                state.status = self
```

Comment:
these 2 should be moved to `kivy_state`, or even better somewhere closer to what it's supposed to do